src/Build/BuildCheck/Infrastructure/CheckScopeClassifier.cs

```diff
@@ -7,8 +7,26 @@
 using Microsoft.Build.Shared;
 
 namespace Microsoft.Build.BuildCheck.Infrastructure;
+
 internal static class CheckScopeClassifier
 {
+    static CheckScopeClassifier() =>
+        FileClassifier.Shared.OnImmutablePathsInitialized += () =>
+        {
+            NotifyOnScopingReadiness?.Invoke();
+
+            FileClassifier.Shared.OnImmutablePathsInitialized -= () => NotifyOnScopingReadiness?.Invoke();
+        };
```

Comment:
I think that this is very hard to read right now. Could you please simplify the coding style?